lib/python3.12/tenants/src/dependency.py

```diff
@@ -0,0 +1,185 @@
+import logging
+import os
+from typing import Any, Dict, Optional, Union
+
+import jwt
+from fastapi import HTTPException, Request, status
+from fastapi.openapi.models import HTTPBearer
+from fastapi.security.base import SecurityBase
+from fastapi.security.utils import get_authorization_scheme_param
+
+from .schema import SupportedAlgorithms, TenantData
+
+logger = logging.getLogger(__name__)
+logger.setLevel(os.getenv("LOG_LEVEL", "INFO"))
+
+
+class TenantDependencyBase:
+    def __init__(
+        self,
+        key: str = "",
+        algorithm: str = "RS256",
+        # TODO: Fix type from bagerdoc to badgerdoc
+        url: str = "http://bagerdoc-keycloack",
```

Comment:
IMHO we need to fix this typo
```suggestion
        url: str = "http://bagerdoc-keycloak",
```